pkg/gardenlet/operation/botanist/operatingsystemconfig.go

```diff
@@ -66,15 +66,16 @@ func (b *Botanist) DefaultOperatingSystemConfig() (operatingsystemconfig.Interfa
 			KubernetesVersion: b.Shoot.KubernetesVersion,
 			Workers:           b.Shoot.GetInfo().Spec.Provider.Workers,
 			OriginalValues: operatingsystemconfig.OriginalValues{
-				ClusterDomain:       gardencorev1beta1.DefaultDomain,
-				Images:              oscImages,
-				KubeletConfig:       b.Shoot.GetInfo().Spec.Kubernetes.Kubelet,
-				MachineTypes:        b.Shoot.CloudProfile.Spec.MachineTypes,
-				SSHAccessEnabled:    v1beta1helper.ShootEnablesSSHAccess(b.Shoot.GetInfo()),
-				ValitailEnabled:     valitailEnabled,
-				ValiIngressHostName: valiIngressHost,
-				NodeLocalDNSEnabled: v1beta1helper.IsNodeLocalDNSEnabled(b.Shoot.GetInfo().Spec.SystemComponents),
-				PrimaryIPFamily:     b.Shoot.GetInfo().Spec.Networking.IPFamilies[0],
+				ClusterDomain:          gardencorev1beta1.DefaultDomain,
+				Images:                 oscImages,
+				KubeletConfig:          b.Shoot.GetInfo().Spec.Kubernetes.Kubelet,
+				MachineTypes:           b.Shoot.CloudProfile.Spec.MachineTypes,
+				SSHAccessEnabled:       v1beta1helper.ShootEnablesSSHAccess(b.Shoot.GetInfo()),
+				ValitailEnabled:        valitailEnabled,
+				ValiIngressHostName:    valiIngressHost,
+				NodeLocalDNSEnabled:    v1beta1helper.IsNodeLocalDNSEnabled(b.Shoot.GetInfo().Spec.SystemComponents),
+				NodeMonitorGracePeriod: *b.Shoot.GetInfo().Spec.Kubernetes.KubeControllerManager.NodeMonitorGracePeriod,
```

Comment:
This can be `nil`, no? Shouldn't we check?

Reply:
As discussed offline, `spec.kubernetes.kubeControllerManager.nodeMonitorGracePeriod` is defaulted via admission plugin (see https://github.com/gardener/gardener/blob/e87e1abce8804016f9ba5f74ddf2ffb32bece865/plugin/pkg/shoot/validator/admission.go#L847). While it is possible to perform a `nil` check here we would also need to either replicate or call the defaulting logic to get a proper value for `NodeMonitorGracePeriod`.
We agreed to keep it as for now.